core/dbt/graph/selector_methods.py

```diff
@@ -431,19 +436,27 @@ def search(self, included_nodes: Set[UniqueId], selector: str) -> Iterator[Uniqu
             resource_type = NodeType(selector)
         except ValueError as exc:
             raise DbtRuntimeError(f'Invalid resource_type selector "{selector}"') from exc
-        for node, real_node in self.parsed_nodes(included_nodes):
-            if real_node.resource_type == resource_type:
-                yield node
+        for unique_id, node in self.parsed_and_unit_nodes(included_nodes):
+            if node.resource_type == resource_type:
+                yield unique_id
 
 
 class TestNameSelectorMethod(SelectorMethod):
     __test__ = False
 
     def search(self, included_nodes: Set[UniqueId], selector: str) -> Iterator[UniqueId]:
-        for node, real_node in self.parsed_nodes(included_nodes):
-            if real_node.resource_type == NodeType.Test and hasattr(real_node, "test_metadata"):
-                if fnmatch(real_node.test_metadata.name, selector):  # type: ignore[union-attr]
-                    yield node
+        command = get_command_name()
+        # Is this too kludgey?
+        if command == "unit-test":
```

Comment:
It may also be desirable to use this selector for other commands, such as `list`. Thinking through the issues with searching for elements in `manifest.unit_tests` in addition to `self.parsed_nodes` as opposed to conditionally based on the commands: 
* duplicate names across unit tests and other TestNode resources
  * this actually feels okay given we are yielding the full unique_id which will be disambiguated by resource type
* selecting nodes that aren't included in the graph queue for commands other than `unit-test`
  * is there a precedent for something like this? perhaps ephemeral models?
  * I think if we can handle this case gracefully by omitting selected unit test cases from the graph queue, this should be okay.

Reply:
The UnitTestCases in manifest.unit_tests had to be included in the graph in order for selection to work. I have been thinking that it ought to work okay to include for other cases because of filter_selection (the task node_selector objects).

Reply:
> thinking that it ought to work okay to include for other cases
same here - could we validate this by removing the `if` statement on the command and instead looping over both `manifest.unit_tests` and `self.parsed_nodes` (or parsed_and_unit_nodes) here instead?